plugins/toolbox-search/src/block_searcher.ts

```diff
@@ -27,29 +27,44 @@ export class BlockSearcher {
     const blockCreationWorkspace = new Blockly.Workspace();
     blockTypes.forEach((blockType) => {
       const block = blockCreationWorkspace.newBlock(blockType);
-      this.indexBlockText(blockType.replaceAll('_', ' '), blockType);
-      block.inputList.forEach((input) => {
-        input.fieldRow.forEach((field) => {
-          this.indexDropdownOption(field, blockType);
-          this.indexBlockText(field.getText(), blockType);
+      const blockState = Blockly.serialization.blocks.save(block);
+      if (blockState != null) {
+        this.indexBlockText(blockType.replaceAll('_', ' '), blockState);
+        block.inputList.forEach((input) => {
+          input.fieldRow.forEach((field) => {
+            this.indexDropdownOption(field, blockState);
+            this.indexBlockText(field.getText(), blockState);
+          });
         });
-      });
+      }
```

Comment:
Instead of wrapping this could you use an early return?

```
if (!blockState) return;
this.indexBlockText // etc...
```

This function already has a lot of nesting and I'd prefer to avoid adding more if we can :P

Reply:
Agree, I will fix it